datefinder.py

```diff
@@ -155,11 +156,20 @@ def _find_and_replace(self, date_string, captures):
         # add timezones to replace
         cloned_replacements = copy.copy(self.REPLACEMENTS)  ## don't mutate
         for tz_string in captures.get('timezones', []):
-            cloned_replacements.update({tz_string: ''})
+            cloned_replacements.update({tz_string: ' '})
 
         date_string = date_string.lower()
         for key, replacement in cloned_replacements.items():
-            date_string = re.sub(key,replacement, date_string, flags=re.IGNORECASE)
+            # we really want to match all permutations of the key surrounded by whitespace chars except one
+            # for example: consider the key = 'to'
+            # 1. match 'to '
+            # 2. match ' to'
+            # 3. match ' to '
+            # but never match r'(\s|)to(\s|)' which would make 'october' > 'ocber'
+            date_string = re.sub(r'(\s|)'+key+'\s',replacement, date_string, flags=re.IGNORECASE)
+            date_string = re.sub(r'\s'+key+'(\s|)',replacement, date_string, flags=re.IGNORECASE)
+            date_string = re.sub(r'\s'+key+'\s',replacement, date_string, flags=re.IGNORECASE)
```

Comment:
There has to be a way to do this with a single replace I think. key has to either be at the beginning of a string, end of a string, or have a space on either side, yes? Testing in the console, I believe this will work:

`r'(^|\s)' + key + '(\s|$)'`